tests/test_views.py

```diff
@@ -51,20 +51,30 @@ def test_survey_edit_observational(client, user, survey_form_entry_observational
 
 
 @pytest.mark.django_db
-def test_survey_publish(client, survey_form_entry):
-    assert not survey_form_entry.published
+def test_survey_preview(client, user, survey_form_entry_observational):
+    client.force_login(user)
+    url = reverse('fobi.view_form_entry', kwargs={'form_entry_slug': survey_form_entry_observational.slug})
+    response = client.get(url)
+
+    assert response.status_code == 200
+    assert not response.context['form_entry'].surveyformentry.published
```

Comment:
Perhaps we could add a check here to make sure that the title of the page is correct? Something like:

```python
assert ('Preview ' + survey_form_entry_observational.name) in response.content.decode('utf-8')
```

Reply:
Yes! I wanted to do this but wasn't sure how to access the page title. Thanks for this tip.